src/fastapi_poe/base.py

```diff
@@ -629,6 +645,263 @@ def make_prompt_author_role_alternated(
 
         return new_messages
 
+    async def capture_cost(
+        self,
+        request: QueryRequest,
+        amounts: Union[List[CostItem], CostItem],
+        access_key: Optional[str] = None,
+        base_url: str = "https://api.poe.com/",
+    ) -> None:
+        """
+
+        Used to capture a cost for monetized creators.
+
+        #### Parameters:
+        - `request` (`QueryRequest`): The currently handlded QueryRequest object.
+        - `amounts` (`Union[list[CostItem], CostItem]`): The to be captured amounts.
+        - `access_key` (`str`): The access_key corresponding to your bot. This is needed to ensure
+        that capture requests are coming from an authorized source.
+
+        """
+
+        if self.access_key:
+            if access_key:
+                warnings.warn(
+                    "Bot already has an access key, access_key parameter is not needed.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
+                request_access_key = access_key
+            else:
+                request_access_key = self.access_key
+        else:
+            if access_key is None:
+                raise InvalidParameterError(
+                    "access_key parameter is required if bot is not"
+                    + " provided with an access_key when make_app is called."
+                )
+            request_access_key = self.access_key
+
+        if not request.bot_query_id:
+            raise InvalidParameterError(
+                "bot_query_id is required to make cost requests."
+            )
+
+        url = f"{base_url}bot/cost/{request.bot_query_id}/capture"
+        result = await self._cost_requests_inner(
+            amounts=amounts, access_key=request_access_key, url=url
+        )
+        if not result:
+            raise InsufficientFundError()
+
+    async def authorize_cost(
+        self,
+        request: QueryRequest,
+        amounts: Union[List[CostItem], CostItem],
+        access_key: Optional[str] = None,
+        base_url: str = "https://api.poe.com/",
+    ) -> None:
+        """
+
+        Used to authorize a cost for monetized creators.
+
+        #### Parameters:
+        - `request` (`QueryRequest`): The currently handlded QueryRequest object.
+        - `amounts` (`Union[list[CostItem], CostItem]`): The to be authorized amounts.
+        - `access_key` (`str`): The access_key corresponding to your bot. This is needed to ensure
+        that authorize requests are coming from an authorized source.
+
+        """
+
+        if self.access_key:
+            if access_key:
+                warnings.warn(
```

Comment:
same here. would prefer removing access key from param and forcing it on bot level